File: breaking-records/breaking-records.go
package breakingrecords

func breakingRecords(scores []int32) []int32 {
	max := scores[0]
	min := scores[0]
	nMaxChanges := 0
	nMinChanges := 0

	for _, s := range scores {
		if s > max {
			max = s
			nMaxChanges++
		}

		if s < min {
			min = s
			nMinChanges++
		}
	}

	return []int32{int32(nMaxChanges), int32(nMinChanges)}
}

File: breaking-records/BreakingRecords.java
import java.util.List;

public class BreakingRecords {

  public static List<Integer> breakingRecords(List<Integer> scores) {
    int max = scores.get(0);
    int min = max;
    int nMaxChanges = 0;
    int nMinChanges = 0;

    for (int i = 0; i < scores.size(); i++) {
      int score = scores.get(i);
      if (score > max) {
        max = score;
        nMaxChanges++;
      } else if (score < min) {
        min = score;
        nMinChanges++;
      }
    }

    return List.of(nMaxChanges, nMinChanges);
  }

}
